etna/models/nn/utils.py

```diff
@@ -236,7 +236,33 @@ def fit(self, ts: TSDataset):
             raise ValueError("Trainer or model is None")
         return self
 
+    def _get_first_prediction_timestamp(self, ts: TSDataset, horizon: int) -> pd.Timestamp:
+        return ts.index[-horizon]
+
+    def _is_in_sample_prediction(self, ts: TSDataset, horizon: int) -> bool:
+        first_prediction_timestamp = self._get_first_prediction_timestamp(ts=ts, horizon=horizon)
+        return first_prediction_timestamp <= self._last_train_timestamp
+
+    def _is_prediction_with_gap(self, ts: TSDataset, horizon: int) -> bool:
+        first_prediction_timestamp = self._get_first_prediction_timestamp(ts=ts, horizon=horizon)
+        first_timestamp_after_train = pd.date_range(self._last_train_timestamp, periods=2, freq=self._freq)[-1]
+        return first_prediction_timestamp > first_timestamp_after_train
+
     def _make_target_prediction(self, ts: TSDataset, horizon: int) -> Tuple[TSDataset, DataLoader]:
+        if self._is_in_sample_prediction(ts=ts, horizon=horizon):
+            raise NotImplementedError(
+                "It is not possible to make in-sample predictions with DeepAR model! "
+                "In-sample predictions aren't supported by current implementation."
+            )
+        elif self._is_prediction_with_gap(ts=ts, horizon=horizon):
```

Comment:
As i understand we should be able to forecast with gap

Reply:
In theory, yes. On practice it isn't currently working. I haven't studied it very thoroughly, but tests just doesn't work as expected. So I decided to add this fence.